### PR TITLE
Full coverage kv management methods

### LINKED ISSUE #510: Ensure full coverage of KV management methods
* https://github.com/nats-io/nats-architecture-and-design/issues/284

### PR DESCRIPTION
**Pull request by Ivandemidov00**
* resolves #510

### CODE REVIEW

**Comment by Ivandemidov00** [comment]
@mtmk I couldn't find the CreateOrUpdateStream api. Do I correctly understand that it is impossible to implement the CreateOrUpdateStore method without calling the server twice? Is CreateOrUpdateStore necessary under such conditions?

**Comment by mtmk** [comment]
> @mtmk I couldn't find the CreateOrUpdateStream api. Do I correctly understand that it is impossible to implement the CreateOrUpdateStore method without calling the server twice? Is CreateOrUpdateStore necessary under such conditions?

Well spotted. let's leave that out. afaik there is no create-or-update stream call. cc @Jarema

File: tests/NATS.Client.KeyValueStore.Tests/KeyValueContextTest.cs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        var cts = new CancellationTokenSource(TimeSpan.FromSeconds(10));
+        var cancellationToken = cts.Token;
+
+        await using var server = NatsServer.StartJS();
+        await using var nats = server.CreateClientConnection();
+
+        var js = new NatsJSContext(nats);
+        var kv = new NatsKVContext(js);
+
+        var natsKVConfig = new NatsKVConfig(bucketName) { Compression = true };
```
Comment (mtmk) [inline tests/NATS.Client.KeyValueStore.Tests/KeyValueContextTest.cs:113]: Compression isn't supported in server 2.9.x which we're still testing against so it's failing the build. maybe use some other property e.g. max_msgs here.

Comment (Ivandemidov00) [reply tests/NATS.Client.KeyValueStore.Tests/KeyValueContextTest.cs]: Fixed. Strange, now fallen another `Reconnect_doesnt_drop_partially_sent_msgs `

Comment (mtmk) [reply tests/NATS.Client.KeyValueStore.Tests/KeyValueContextTest.cs]: don't worry. that's a flapper. I'll re-run it.

File: src/NATS.Client.KeyValueStore/NatsKVContext.cs [hunk 2]
```diff
@@ -19,6 +19,8 @@ public enum NatsKVStorageType
 /// </summary>
 public class NatsKVContext : INatsKVContext
 {
+    private const string KvStreamPrefix = "KV_";
```
Comment (rickdotnet) [inline src/NATS.Client.KeyValueStore/NatsKVContext.cs:22]: Generally, wouldn't be a stickler on this but using `KvStreamNamePrefix` would separate it from the subject prefix `$KV` used below.

File: src/NATS.Client.KeyValueStore/NatsKVContext.cs [hunk 3]
```diff
@@ -45,10 +49,130 @@ public ValueTask<INatsKVStore> CreateStoreAsync(string bucket, CancellationToken
...
+    internal static void ValidateBucketName(string bucket)
+    {
+        if (!ValidBucketRegex.IsMatch(bucket))
+        {
+            throw new NatsKVException("Invalid bucket name");
+        }
+    }
+
+    private static string BucketToStream(string bucket)
+        => $"KV_{bucket}";
```
Comment (rickdotnet) [inline src/NATS.Client.KeyValueStore/NatsKVContext.cs:169]: I'd suggest using `KvStreamNamePrefix` here.

**Review by rickdotnet — CHANGES_REQUESTED** [review]
Aside from two minor suggestions, this looks great. I can make those updates if you'd like.

**Comment by mtmk** [comment]
@Ivandemidov00 can you sign your commits next time please? (if you're comfortable with git rebase you can push -f that would be good but don't worry if not) sorry keep forgetting to remind you and you've contributed quite a bit already, we're very grateful for that 💯

**Review by mtmk — APPROVED** [review]

**Pull request merged**